keps/sig-api-machinery/3488-cel-admission-control/README.md

```diff
@@ -1023,26 +1025,139 @@ this enhancement.
 
 #### Enforcement Actions
 
-For phase 1, all violations implicitly result in a `deny` enforcement action.
+`ValidatingAdmissionPolicyBinding` resources may control how admission is
+enforced. This is performed using a single field. E.g.:
 
-For phase 2, we intend to support multiple enforcement actions.
+```yaml
+apiVersion: admissionregistration.k8s.io/v1alpha1
+kind: ValidatingAdmissionPolicyBinding
+...
+spec:
+  validationActions: [warn, audit] # optional field
+```
 
-Use cases:
+The enum options will be:
+
+- `deny`: Validation failures result in a denied request. (default beahvior if
+  field is unset)
```

Comment:
if we're going to default, `deny` is probably the right default, but is it better to default or make this an explicit decision the binding author has to make?

Reply:
I was going to say default it better, but we have a wrinkle where if we don't require an explicit decision, then `validationActions: []` will be different than when `validationActions` is unset. By making it an explicit decision we remove that case and make things very clear. I'll update.

Reply:
> then validationActions: [] will be different than when validationActions is unset.

unless you make the request using protobuf, which loses the distinction between `[]` and `null` 🙄 

dropping that edge seems good